### PR TITLE
Rename ProjectSettings macros to follow the singleton's rename

### PR DESCRIPTION
**Pull request by Calinou**
The Globals singleton was renamed to ProjectSettings in 3.0. Additionally, this renames some of the macros to be more explicit (including for EditorSettings).

`_DEF` was renamed to `_DEFAULT` to avoid having the same length as `_GET` (which can look confusing at a glance).

This also adds comments above each macro, which will appear in many IDEs when hovering the macro names.

See https://github.com/godotengine/godot/issues/16863#issuecomment-871572390.

### CODE REVIEW

File: core/config/project_settings.cpp [hunk 1]
```diff
@@ -510,19 +510,19 @@ Error ProjectSettings::_setup(const String &p_path, const String &p_main_pack, b
 Error ProjectSettings::setup(const String &p_path, const String &p_main_pack, bool p_upwards) {
 	Error err = _setup(p_path, p_main_pack, p_upwards);
 	if (err == OK) {
-		String custom_settings = GLOBAL_DEF("application/config/project_settings_override", "");
+		String custom_settings = PROJECT_DEFAULT("application/config/project_settings_override", "");
```
Comment (reduz) [inline core/config/project_settings.cpp:513]: GLOBAL_DEF is not short for GLOBAL_DEFAULT, its GLOBAL_DEFINE.

**Comment by reduz** [comment]
I am not sure myself, not against a rename, but what is proposed does not seem better than what it is already there. Maybe PROJECT_SETTING_DEF and PROJECT_SETTING_DEF
EDITOR_SETTING_DEF and EDITOR_SETTING_GET

File: core/config/project_settings.cpp (hunk 1 above)
Comment (aaronfranke) [reply core/config/project_settings.cpp:513]: I suppose this is a good argument for renaming it then.

Comment (EricEzaM) [reply core/config/project_settings.cpp:513]: Haha I always thought it was define, not default, and was a bit confused reading the OP.

**Comment by EricEzaM** [comment]
As a reminder, having project settings being able to be defined anywhere does not work so well with the doctool:

https://github.com/godotengine/godot-proposals/issues/1339#issuecomment-671337161 and https://github.com/godotengine/godot/pull/48548#discussion_r628672846

> Also, as a result of the EditorNode and related classes not being constructed [when running doctool], some project settings are actually left out of the documentation, such as editor/main_run_args which has it's GLOBAL_DEF in the EditorNode constructor, so when running the doctool it is never called.

Same problem occurs with EditorSettings.

**Comment by reduz** [comment]
@EricEzaM the general idea is that they are defined on initialization, but having them one in a place is quite messy, I prefer to have them closer to where they are relevant if possible.

**Comment by aaronfranke** [comment]
@reduz What is messy about it? We already do this with editor settings.

**Comment by mhilbrunner** [comment]
PR meeting: DEFAULT -> DEFINE

Also we're not against this, but we maybe should take the opportunity to modernise this
(@akien-mga, @aaronfranke @reduz had input, so maybe discuss further with them)

**Pull request closed without merging**